src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2128,9 +2143,25 @@ public boolean waitForSchemaAgreement(long maxWait, TimeUnit unit, BooleanSuppli
         }
     }
 
-    public boolean haveMajorVersion3Nodes()
+    /**
+     * Returns {@code true} only if the information about the version of each node in the cluster is available and
+     * ALL the nodes are on 4.0 (regardless of the patch version).
```

Comment:
The comment does not seem to describe the method correctly. I think you want the below according to the current behavior. 

```suggestion
     * Returns {@code false} only if the information about the version of each node in the cluster is available and
     * ALL the nodes are on 4.0 (regardless of the patch version).
```

Reply:
indeed, thanks